tpm2tools/signer.go

```diff
@@ -0,0 +1,81 @@
+package tpm2tools
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/google/go-tpm/tpm2"
+)
+
+var (
+	// Global mutex to protect against concurrent TPM read/writes.
+	signerMutex = &sync.Mutex{}
+)
+
+// TpmSigner implements the crypto.Signer interface for TPM Keys.
+// Concurrent use of one or more TpmSigners is thread safe, but it is not safe
+// to read/write to the TPM from other sources while using a TpmSigner.
+type TpmSigner struct {
+	key *Key
+}
+
+// Public returns the TpmSigners public key.
+func (signer TpmSigner) Public() crypto.PublicKey {
```

Comment:
If we add another member to `TpmSigner`, Here and elsewhere, we should use pointer recivers for the methods, as that's more typical for `crypto.Signer` implementations.

Reply:
I don't really see a reason to use a pointer receiver, since TpmSigner is essentially already a pointer to a Key and crypto.Hash is just a uint so not much additional overhead.